base/src/pos.rs

```diff
@@ -118,14 +118,69 @@ impl fmt::Display for Location {
     }
 }
 
+/// An expansion identifier tracks wheter a span originated from a macro expansion or not
+#[derive(Copy, Clone, Default, Ord, PartialOrd, Eq, PartialEq, Debug)]
+pub struct ExpansionId(pub u32);
+
+pub const NO_EXPANSION: ExpansionId = ExpansionId(0);
+
+pub const UNKNOWN_EXPANSION: ExpansionId = ExpansionId(1);
```

Comment:
Mimicked how Rust uses this field, e could later on generate `ExpansionId`s for each expanded macro so

```
import "asd" // ExpansionId(2)
import "asd" // ExpansionId(3)
some_other_macro "asd" // ExpansionId(4)
```